net/pfSense-pkg-WireGuard/files/usr/local/pkg/wireguard/wg_guiconfig.inc

```diff
@@ -0,0 +1,338 @@
+
+	}
+
+	return null;
+
+}
+
+function wg_generate_tunnel_address_popup_link($tunnel_name) {
+	global $wgg;
+
+	$tun_idx = wg_get_tunnel_array_index($tunnel_name);
+
+	if (isset($wgg['tunnels'][$tun_idx]) && is_array($wgg['tunnels'][$tun_idx])) {
+
+		$tunnel = $wgg['tunnels'][$tun_idx];
+
+		$addresses = $tunnel['addresses']['row'];
+
+		if (!is_wg_tunnel_assigned($tunnel['name'])) {
+			
+			if (is_array($addresses) && count($addresses) > 0) {
+
+				$href = "vpn_wg_tunnels_edit.php?tun={$tunnel['name']}";
+
+				$extras = count($addresses) - 1;
+
+				if ($extras > 0) {
+
+					$data_html = "<table><thead><th>Address</th><th style='padding-left: 10px;'>Description</th></thead><tbody>";
+		
+					foreach ($addresses as $address) {
+		
+						$data_html .= "<tr><td>{$address['address']}/{$address['mask']}</td><td style='padding-left: 10px;'>{$address['descr']}</td></tr>";
+		
+					}
+		
+					$data_html .= '</tbody></table>';
+		
+					$hint = "{$addresses[0]['address']} (+{$extras})";
+		
+					$ret_html = "<a href=\"{$href}\" title=\"Interface Addresses\" data-toggle=\"popover\" data-trigger=\"hover focus\" data-content=\"{$data_html}\" data-html=\"true\">{$hint}</a>";
+		
+					return $ret_html;
+		
+				} else {
+		
+					$ret_html = "<a href=\"{$href}\" title=\"{$addresses[0]['descr']}\">{$addresses[0]['address']}/{$addresses[0]['mask']}</a>";
+		
+					return $ret_html;
+				
+				}
+		
+			} else {
+
+				return '(none)';
+
+			}
+
+		} else {
+
+			$wg_pfsense_if = wg_get_pfsense_interface_info($tunnel['name']);
+
+			return "<i class=\"fa fa-sitemap\" style=\"vertical-align: middle;\"></i><a style=\"padding-left: 3px\" href=\"../../interfaces.php?if={$wg_pfsense_if['name']}\">{$wg_pfsense_if['descr']} ({$wg_pfsense_if['name']})</a>";
+
+		}
+
+	}
+
+}
+
+function wg_generate_peer_allowedips_popup_link($peer_idx) {
+	global $wgg;
+
+	if (isset($wgg['peers'][$peer_idx]) && is_array($wgg['peers'][$peer_idx])) {
+
+		$peer = $wgg['peers'][$peer_idx];
+
+		$allowedips = $peer['allowedips']['row'];
+			
+		if (is_array($allowedips) && count($allowedips) > 0) {
+
+			$href = "vpn_wg_peers_edit.php?peer={$peer_idx}";
+
+			$extras = count($allowedips) - 1;
+
+			if ($extras > 0) {
+
+				$data_html = "<table><thead><th>Subnet</th><th style='padding-left: 10px;'>Description</th></thead><tbody>";
+	
+				foreach ($allowedips as $address) {
+	
+					$data_html .= "<tr><td>{$address['address']}/{$address['mask']}</td><td style='padding-left: 10px;'>{$address['descr']}</td></tr>";
+	
+				}
+	
+				$data_html .= '</tbody></table>';
+	
+				$hint = "{$allowedips[0]['address']} (+{$extras})";
+	
+				$ret_html = "<a href=\"{$href}\" title=\"Allowed IPs\" data-toggle=\"popover\" data-trigger=\"hover focus\" data-content=\"{$data_html}\" data-html=\"true\">{$hint}</a>";
+	
+				return $ret_html;
+	
+			} else {
+	
+				$ret_html = "<a href=\"{$href}\" title=\"{$allowedips[0]['descr']}\">{$allowedips[0]['address']}/{$allowedips[0]['mask']}</a>";
```

Comment:
The contents of these variables should be encoded with `htmlspecialchars()` before printing.